src/web/rustdoc.rs

```diff
@@ -447,7 +448,7 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
         (target, inner_path.join("/"))
     };
 
-    // Find the path of the latest version for the `Go to latest` and `Permalink` links
+    // For `Permalink`, the path of the latest version, with an explicit version in it.
     let mut latest_path = if latest_release.build_status {
```

Comment:
This looks wrong. This code is not calculating a link to the *current* page, it's finding the page that "Go to latest" should link to. So I think the Permalink feature you already implemented doesn't work properly.

Reply:
I don't understand that. This code as I understands it links (via target-redirect) to what the _current_ page would be, if you substituted the latest numeric version into the URL. That used to be useful for "Go to latest version," but now it's useful for "Permalink."

Reply:
... right, because "Permalink" is only used when you're already on the latest page. Ok, that makes sense to me.